client/my-sites/comments/comment-list/index.jsx

```diff
@@ -93,7 +103,11 @@ export class CommentList extends Component {
 
 	isCommentSelected = commentId => !! this.state.selectedComments[ commentId ];
 
-	isSelectedAll = () => this.props.comments.length === size( this.state.selectedComments );
+	isSelectedAll = () => this.getComments().length === size( this.state.selectedComments );
+
+	removeFromPersistedComments = commentId => this.setState( {
+		persistedComments: reject( this.state.persistedComments, { ID: commentId } ),
+	} );
```

Comment:
I just recently became aware of the function-passing syntax here and we should probably use it because of the way React can batch state updates. that is, because we're accessing the previous state we should pass in a function…

```js
this.setState( ( { persistedComments } ) => ( {
	persistedComments: reject( persistedComments, { ID: commentId } ),
} ) );
```